### ISSUE TITLE
how to evaluate the map using coco metrix in custom dataset?

### ISSUE THREAD

**Issue by luoshiyong** [open-mmlab/mmdetection#12323]
``i  had implement a custom dataset in instance segmentation ,which labels is  labelme json,  but in mmdet ,there is no evaluation metric for labelme json, so i modified the coco metrix as follows:  `val_evaluator = dict(
    type='CisdiMetric',
    ann_file=None,
    metric=['bbox', 'segm'],
    format_only=False,
    backend_args=backend_args)
test_evaluator = val_evaluator`   , and in metrics, i use the following code to get the `gt,if self._coco_api is None:
                # TODO: Need to refactor to support LoadAnnotations
                assert 'gt_instances' in data_sample, \
                    'ground truth is required for evaluation when ' \
                    '`ann_file` is not provided'
                gt['anns'] = []
                # print(data_sample)
                boxes = data_sample['gt_instances']['bboxes'].detach().cpu().numpy()
                labels = data_sample['gt_instances']['labels'].detach().cpu().numpy()
                masks = [] if data_sample['gt_instances'].get('masks',None) is None else data_sample['gt_instances']['masks'].raw_masks
                # print(data_sample['gt_instances']['masks'])
                for bbox, label, mask in zip(boxes, labels, masks):
                    gt['anns'].append({'bbox': bbox, 'bbox_label': label,"mask":mask})
                    # gt['anns'] = data_sample['gt_instances']
            # add converted result to the results list
            self.results.append((gt, result))`,  but the result is all map is 0,

**Comment by luoshiyong** [comment]
`if self._coco_api is None:
                # TODO: Need to refactor to support LoadAnnotations
                assert 'gt_instances' in data_sample, \
                    'ground truth is required for evaluation when ' \
                    '`ann_file` is not provided'
                gt['anns'] = []
                # print(data_sample)
                boxes = data_sample['gt_instances']['bboxes'].detach().cpu().numpy()
                labels = data_sample['gt_instances']['labels'].detach().cpu().numpy()
                masks = [] if data_sample['gt_instances'].get('masks',None) is None else data_sample['gt_instances']['masks'].raw_masks
                # print(data_sample['gt_instances']['masks'])
                for bbox, label, mask in zip(boxes, labels, masks):
                    gt['anns'].append({'bbox': bbox, 'bbox_label': label,"mask":mask})
                    # gt['anns'] = data_sample['gt_instances']
            # add converted result to the results list
            self.results.append((gt, result))`

**Comment by luoshiyong** [comment]
![Image](https://github.com/user-attachments/assets/bcdcdb2b-899a-4a30-8e40-9134411e8e02)

**Comment by yanhuojiyusheng** [comment]
Suggest to convert labelme to coco format this is very well done!
